### PR TITLE
Add show_incorrect_images function

### PR DESCRIPTION
**Pull request by zachwe**
* The main valuable thing for you in this PR is the `show_incorrect_images` function, which will accepts model, x, and y, and will print the before and after images where model predicts incorrectly.
* I also added code (currently commented out) in generate_features which will include the before/after paths in the pickled data
* You can see examples of using `show_incorrect_images` down at the bottom of zach-models

* This PR also includes some experimenting with openCV thresholds for segmenting an image into only bright/not bright spots. I think it could be useful, though a logistic regression using *only* those is worse than useless :|

### CODE REVIEW

File: src/generate_features.py [hunk 1]
```diff
@@ -42,12 +52,16 @@ def baseline_hist_diff(img_before, img_after):
...
+        diff = baseline_hist_diff(file_1, file_2, img_post=binary_threshold)
         img_hist.append(diff)
+        before_paths.append(file_1)
+        after_paths.append(file_2)
 
     column_names = ["hist" + str(i) for i in range(256)]
     df = pd.DataFrame(img_hist, columns=column_names, index=patient_ids)
     df["y"] = Y
+    # df["before_path"] = before_paths
+    # df["after_path"] = after_paths
```
Comment (zachwe) [inline src/generate_features.py]: Uncomment these lines to include the paths in your dataframe.

File: src/generate_features.py [hunk 2]
```diff
@@ -13,14 +13,23 @@
 import utils
 
 
-def hist_features(img_path):
-    img = cv2.imread(img_path, 0)
+def binary_threshold(img):
```
Comment (zachwe) [inline src/generate_features.py]: This whole file is not exactly usable in the sense that you can run it and everything will Just Work but I think the code here is a useful start for building other feature sets.

File: src/utils.py [hunk 3]
```diff
@@ -6,6 +7,27 @@
 
 from sklearn import model_selection
 
+
+def show_incorrect_images(model, x_test, y_test):
```
Comment (zachwe) [inline src/utils.py]: This is the function which I'm hoping will be useful, check out zach-models notebook for some usage

**Review by annamerk — APPROVED** [review]

File: src/generate_features.py [hunk 4]
```diff
@@ -13,14 +13,23 @@
...
     hist = cv2.calcHist([img],[0],None,[256],[0,256])
     return hist.flatten()
 
-def baseline_hist_diff(img_before, img_after):
-    return hist_features(img_before) - hist_features(img_after)
+def baseline_hist_diff(img_before, img_after, img_post=None):
+    return hist_features(img_before, img_post=img_post) - hist_features(img_after, img_post=img_post)
 
 if __name__ == '__main__':
     input_csv = pd.read_csv("data_binary.csv")
```
Comment (annamerk) [inline src/generate_features.py]: If we wanna keep this code copy outside of the other notebook, let's make it a function so the notebook can still call it? then main can just do the saving or something.

**Pull request merged**